### PR TITLE
GO-2300: clean import to existing space

### PR DESCRIPTION
**Pull request by AnastasiaShemyakinskaya**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### Description

* Add new parameter in protocol for PB import - Gallery or Space imprt
* Separate logic of creation result import collection for Gallery and Space
* For Space import logic is the same as before
* For Gallery logic is implemented according to task: https://linear.app/anytype/issue/GO-2300/clean-import-to-existing-space

### What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Related Tickets & Documents

https://linear.app/anytype/issue/GO-2300/clean-import-to-existing-space
### Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [x] 🙅 no documentation needed

### CODE REVIEW

**Review by fat-fellow — APPROVED** [review]

File: core/block/import/pb/gallery.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+type GalleryImport struct {
+	service *collection.Service
+}
+
+func NewGalleryImport(service *collection.Service) *GalleryImport {
+	return &GalleryImport{service: service}
+}
+
+func (g *GalleryImport) ProvideCollection(_ []*converter.Snapshot, widget *converter.Snapshot, _ map[string]string, params *pb.RpcObjectImportRequestPbParams, workspaceSnapshot *converter.Snapshot) (*converter.Snapshot, error) {
```
Comment (KirillSto) [inline core/block/import/pb/gallery.go:26]: Again I suggest here very-short-line-but-long-to-read-advice function length :upside_down_face: 
```
func (g *GalleryImport) ProvideCollection(
    _ []*converter.Snapshot, 
    widget *converter.Snapshot,
     _ map[string]string, 
    params *pb.RpcObjectImportRequestPbParams, 
    workspaceSnapshot *converter.Snapshot,
) (*converter.Snapshot, error) {
```

File: core/block/import/pb/gallery_test.go [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+		collection, err := collectionProvider.ProvideCollection(nil, nil, nil, params, nil)
+
+		// then
+		assert.Nil(t, err)
+		assert.NotNil(t, collection)
+		rootCollectionState := state.NewDocFromSnapshot("", collection.Snapshot).(*state.State)
+		objectsInCollection := rootCollectionState.GetStoreSlice(template.CollectionStoreKey)
+		assert.Len(t, objectsInCollection, 0)
+	})
+	t.Run("CollectionTitle parameter is empty - collection with default name", func(t *testing.T) {
```
Comment (KirillSto) [inline core/block/import/pb/gallery_test.go:36]: **given** and **when** are identical in first and second usecases. Maybe we can unite the assertions?

File: core/block/import/pb/gallery_test.go [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+
+		// when
+		collection, err := collectionProvider.ProvideCollection(nil, nil, nil, params, nil)
+
+		// then
+		assert.Nil(t, err)
+		assert.NotNil(t, collection)
+		assert.Equal(t, rootCollectionName, collection.FileName)
+	})
+	t.Run("CollectionTitle parameter is equeal \"test\" - collection with name test", func(t *testing.T) {
```
Comment (KirillSto) [inline core/block/import/pb/gallery_test.go:49]: equal 'test' ?

File: core/block/import/pb/gallery_test.go [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+	"github.com/gogo/protobuf/types"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/anyproto/anytype-heart/core/block/editor/state"
+	"github.com/anyproto/anytype-heart/core/block/editor/template"
+	"github.com/anyproto/anytype-heart/core/block/editor/widget"
+	"github.com/anyproto/anytype-heart/core/block/import/converter"
+	"github.com/anyproto/anytype-heart/pb"
+	"github.com/anyproto/anytype-heart/pkg/lib/bundle"
+	smartblock2 "github.com/anyproto/anytype-heart/pkg/lib/core/smartblock"
```
Comment (KirillSto) [inline core/block/import/pb/gallery_test.go:15]: No other packages end with smartblock, so smartblock2 could be returned initial name

File: core/block/import/pb/space.go [hunk 5]
```diff
@@ -0,0 +1,110 @@
+package pb
+
+import (
+	"github.com/samber/lo"
+
+	"github.com/anyproto/anytype-heart/core/block/collection"
+	"github.com/anyproto/anytype-heart/core/block/editor/state"
+	widgets "github.com/anyproto/anytype-heart/core/block/editor/widget"
```
Comment (KirillSto) [inline core/block/import/pb/space.go:8]: Widgets could be returned its initial value as well

File: util/builtinobjects/builtinobjects.go [hunk 6]
```diff
@@ -252,7 +252,7 @@ func (b *builtinObjects) inject(ctx session.Context, spaceID string, useCase pb.
 	return
 }
 
-func (b *builtinObjects) importArchive(ctx context.Context, spaceID string, path string, noProgress bool) (err error) {
+func (b *builtinObjects) importArchive(ctx context.Context, spaceID string, path string, noProgress bool, title string, importType pb.RpcObjectImportRequestPbParamsType) (err error) {
```
Comment (KirillSto) [inline util/builtinobjects/builtinobjects.go:255]: Idk why I like this square-like function-definition style

**Review by KirillSto — APPROVED** [review]

**Pull request merged**